### PR TITLE
Added minimum layer definition for Convolutional and Fully Connected

### PR DESCRIPTION
**Pull request by mcrispino**

### CODE REVIEW

File: MLModelInstance.xml [hunk 1]
```diff
@@ -768,5 +760,84 @@
...
+                              ElementType="Layer"
+                              Multiple="true" />
+            </ChildrenElements>
+            <Attributes>
+                <Attribute Name="encoder"
+                           Type="enum{Stacked CNN}"
+                           DefaultValue="Stacked CNN"
+                           Description="Encoder to use"
+                           PrettyName="Encoder">
+                </Attribute>
```
Comment (jechague) [inline MLModelInstance.xml:780]: We should be careful of using terminology from Ludwig that is not widely accepted across the ML community. I know they refer to their network's input layers as "encoders" but that term is generally used when working under a Encoder-Decoding architecture (like in an autoencoder)

File: MLModelInstance.xml [hunk 2]
```diff
@@ -768,5 +760,84 @@
...
+            </Attributes>
+        </ElementType>
+
+        <ElementType Name="Layer"
+                     Caption="{0} Layer"
```
Comment (jechague) [inline MLModelInstance.xml:785]:
```suggestion
                     Caption="{0}"
```
This is the change we discussed via email about dropping the "Layer" part of the name of an instance of a layer

**Review by jechague — CHANGES_REQUESTED** [review]
Marking this as a request for changes although I understand this is WIP

File: MLModelInstance.xml (hunk 1 above)
Comment (mcrispino) [reply MLModelInstance.xml:780]: What name would you suggest?

File: MLModelInstance.xml [hunk 3]
```diff
@@ -768,5 +760,84 @@
...
+                           Description="Pool stride for the Convolutional Layer"
+                           ApplyIf="self::*[@layerType = 'Convolutional']">
+                </Attribute>
+
+                <Attribute Name="fc_size"
+                           Type="int"
+                           DefaultValue="128"
+                           PrettyName="Fully Conneced Size"
+                           Description="Number of nodes of the Fully Connected Layer"
+                           ApplyIf="self::*[@layerType = 'Fully Connected']">
```
[the remark this answers is not in the document]
Comment (mcrispino) [reply MLModelInstance.xml:829]: Would it be the same attribute for both layer types, or could we have different attributes for the different layers?

If it is the same attribute, then the name change is fine. But if each layer type will have its own attribute, we can keep the Ludwig's name...

File: MLModelInstance.xml [hunk 4]
```diff
@@ -768,5 +760,84 @@
...
+                </Attribute>
+                
+                <Attribute Name="dropout"
+                           Type="decimal"
+                           DefaultValue="0.0"
+                           PrettyName="Dropout"
+                           Description="Dropout. Value must be between 0.0 and 1.0">
+                </Attribute>
+
+            </Attributes>
```
[the remark this answers is not in the document]
Comment (mcrispino) [reply MLModelInstance.xml:845]: Fixed at [f01bf41](https://github.com/genexuslabs/mlmodel-pattern/pull/6/commits/f01bf41b900d4faaf341b56b3f71c96e8cd99647)

File: MLModelInstance.xml (hunk 2 above)
Comment (mcrispino) [reply MLModelInstance.xml]: Take a look at commit [f01bf41](https://github.com/genexuslabs/mlmodel-pattern/pull/6/commits/f01bf41b900d4faaf341b56b3f71c96e8cd99647), removed the "layer" part but added the layer's name to the caption.

File: MLModelInstance.xml (hunk 3 above)
[the remark this answers is not in the document]
Comment (jechague) [reply MLModelInstance.xml:835]: They are technically the same concept but named differently depending on whether the layers works on images (where the "filter" name is used) as opposed to tensors of other types (where the "units" name is used). 

The ideal thing would be to have a single attribute whose name changes depending on the layer type but I don't know if that possible here.

Note that the names used for these attributes is not a Ludwig-specific thing but general ML terminology. Furthermore, and as noted on my following comment below, we should minimize, and if possible avoid, using Ludwig-specific terminology

File: MLModelInstance.xml (hunk 1 above)
Comment (jechague) [reply MLModelInstance.xml:780]: Correct me if I'm wrong, but this attribute represent the input layer of the model, does it not?

If that the case, I think they should not have a name at all as this section of the model should not be separate from the other layers (the one defined under "Layer").

File: MLModelInstance.xml (hunk 3 above)
[the remark this answers is not in the document]
Comment (mcrispino) [reply MLModelInstance.xml:835]: If they have different names, we'll need two attributes. This one, `fc_size`, will be available for fully connected layers only, and we can have another attribute, with another name, for LSTMs or other layer types.

File: MLModelInstance.xml (hunk 1 above)
Comment (mcrispino) [reply MLModelInstance.xml]: Ok. Removed the `encoder` attribute at [c75ecf7](https://github.com/genexuslabs/mlmodel-pattern/pull/6/commits/c75ecf7783170e1f1290fddfcbcb83dd1294decf).

File: MLModelInstance.xml (hunk 3 above)
[the remark this answers is not in the document]
Comment (jechague) [reply MLModelInstance.xml:826]: Oh, I think I misunderstood your question: my answer was referring to `num_units` (from FC and LSTM layers) and `num_filters` (from CNNs). For both FC and LSTM layers the parameters is the same: `num_units` so we should have a single parameter for both

[the remark this answers is not in the document]
Comment (mcrispino) [reply MLModelInstance.xml]: Ok, renamed `fc_size` to `num_units`, at [2facd91](https://github.com/genexuslabs/mlmodel-pattern/pull/6/commits/2facd919226326f45529eb49a6c5a58b63c1d558)

**Review by jechague — APPROVED** [review]
LGTM

**Pull request merged**